File: src/builder/java/builder/Builder.java
package builder.java.builder;

import builder.java.champion.Champion;

import static builder.java.champion.Champion.*;

/**
 * Created by kvirus on 2019/6/13 18:49
 * Email @ caoyingkui@pku.edu.cn
 * <p>
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
public interface Builder {
    public Builder setNickname(String nickname);

    public Builder setSlogon(String slogon);

    public Builder setContinent(String continent);

    public Builder setType(ChampionType type);

    public Champion build();
}

File: src/builder/kotlin/builder/Builder.kt
package builder.kotlin.builder

import builder.kotlin.champion.Champion
import builder.kotlin.champion.ChampionType

/**
 * Created by kvirus on 2019/6/13 19:30
 * Email @ caoyingkui@pku.edu.cn
 *
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
interface Builder {
    abstract fun setNickname(nickname: String): Builder
    abstract fun setSlogon(slogon: String): Builder
    abstract fun setContinent(continent: String): Builder
    abstract fun setType(type: ChampionType): Builder
    abstract fun build(): Champion
}